connector/ldap/ldap.go

```diff
@@ -2,20 +2,109 @@
 package ldap
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
+	"log"
 
 	"gopkg.in/ldap.v2"
 
 	"github.com/coreos/dex/connector"
 )
 
-// Config holds the configuration parameters for the LDAP connector.
+// Config holds the configuration parameters for the LDAP connector. The LDAP
+// connectors require executing two queries, the first to find the user based on
+// the username and password given to the connector. The second to use the user
+// entry to search for groups.
+//
+// An example config:
+//
+//     type: ldap
+//     config:
+//       host: ldap.example.com:636
+//       rootCA: /etc/dex/ldap.ca
+//       bindDN: uid=seviceaccount,cn=users,dc=example,dc=com
+//       bindPW: password
+//       userSearch:
+//         # Would translate to the query "(&(objectClass=person)&(uid=<username>))"
+//         baseDN: cn=users,dc=example,dc=com
+//         filter: "(objectClass=person)"
+//         username: uid
```

Comment:
FreeIPA actually uses "uid" as the human readable name you enter when doing binds. For example here's my LDAP entry:

```
dn: uid=echiang,cn=users,cn=accounts,dc=core,dc=example,dc=so
displayName: Eric Chiang
uid: echiang
objectClass: ipaobject
objectClass: person
objectClass: top
objectClass: ipasshuser
objectClass: inetorgperson
objectClass: organizationalperson
objectClass: krbticketpolicyaux
objectClass: krbprincipalaux
objectClass: inetuser
objectClass: posixaccount
objectClass: ipaSshGroupOfPubKeys
objectClass: mepOriginEntry
loginShell: /bin/bash
initials: EC
gecos: Eric Chiang
sn: Chiang
homeDirectory: /home/echiang
givenName: Eric
cn: Eric Chiang
uidNumber: 300015
gidNumber: 300015
```